### PR TITLE
[train_t2i_adapter_sdxl.py] Fix the LR scheduler when num_train_epochs is passed in a distributed training env

### PR DESCRIPTION
**Pull request by adi-IL**
Fixes #8384

Follow-up to #14527 and #14528, still one script. This PR updates `examples/t2i_adapter/train_t2i_adapter_sdxl.py`.

### What was wrong

When `--num_train_epochs` is used, `train_t2i_adapter_sdxl.py` built the LR schedule from the unsharded dataloader and ignored `accelerator.num_processes`, resulting in mismatched warmup and total training steps across distributed environments.

### What changed

Applies the #8312 pattern already established across the DreamBooth, ControlNet, and Text-to-Image trainers:

- Warmup and training steps passed to `get_scheduler` are scaled by `accelerator.num_processes`.
- Step counts used to build the schedule account for the sharded dataloader length (`math.ceil(len(train_dataloader) / accelerator.num_processes)`).
- After `accelerator.prepare`, a warning is logged if the prepared dataloader length deviates from that expectation.
- Added `test_t2i_adapter_sdxl_num_train_epochs` in `examples/t2i_adapter/test_t2i_adapter.py` to cover the `--num_train_epochs` path.

### Coordination

Covered by the ongoing community tracker #8384. cc @sayakpaul @geniuspatrick

### Minimal training command using `num_train_epochs`

```bash
export MODEL_DIR="stabilityai/stable-diffusion-xl-base-1.0"
export OUTPUT_DIR="sdxl-t2i-adapter-model"

accelerate launch train_t2i_adapter_sdxl.py \
  --pretrained_model_name_or_path=$MODEL_DIR \
  --output_dir=$OUTPUT_DIR \
  --dataset_name=fusing/fill50k \
  --mixed_precision="fp16" \
  --resolution=1024 \
  --learning_rate=1e-5 \
  --num_train_epochs=2 \
  --train_batch_size=1 \
  --gradient_accumulation_steps=4 \
  --seed=42
```

### Tests I ran

```bash
pytest examples/t2i_adapter/test_t2i_adapter.py
python3 -m py_compile examples/t2i_adapter/train_t2i_adapter_sdxl.py examples/t2i_adapter/test_t2i_adapter.py
ruff check examples/t2i_adapter/train_t2i_adapter_sdxl.py examples/t2i_adapter/test_t2i_adapter.py
ruff format --check examples/t2i_adapter/train_t2i_adapter_sdxl.py examples/t2i_adapter/test_t2i_adapter.py
```

All passed (`2 passed in 84.45s`).

### CODE REVIEW

**Comment by adi-IL** [comment]
Hi @sayakpaul, gentle check-in on this LR scheduler fix for train_t2i_adapter_sdxl.py in distributed training settings. Let me know if you would like any adjustments.

**Comment by adi-IL** [comment]
Closing in favor of the umbrella PR that packs the same LR-scheduler `num_train_epochs` fix across all of these example scripts: https://github.com/huggingface/diffusers/pull/14743

That PR supersedes #14527 #14528 #14540 #14546 #14597 so review can happen once. Tip contents were ported onto current `main` there.
